internal/uvm/create_lcow.go

```diff
@@ -298,11 +302,34 @@ func CreateLCOW(ctx context.Context, opts *OptionsLCOW) (_ *UtilityVM, err error
 				ImageFormat: imageFormat,
 			},
 		}
-		// Add to our internal structure
-		uvm.vpmemDevices[0] = &vpmemInfo{
-			hostPath: opts.RootFSFile,
-			uvmPath:  "/",
-			refCount: 1,
+		if uvm.vpmemMultiMapping {
+			pmem := newPackedVPMemDevice()
+			pmem.maxMappedDeviceCount = 1
+
+			st, err := os.Stat(rootfsFullPath)
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to stat rootfs: %q", rootfsFullPath)
+			}
+			devSize := pageAlign(uint64(st.Size()))
+			memReg, err := pmem.Allocate(devSize)
+			defer func() {
+				if err != nil {
+					err = pmem.Release(memReg)
+					log.G(ctx).WithError(err).Debug("failed to release memory region")
```

Comment:
shouldn't this be inside an if for when the release function fails? Otherwise we'll run this every time there's a failure of the func even if the release operation was successful.